File: internal/bot/command/list_group.go
package command

import (
	"fmt"
	"strings"

	tgbotapi "github.com/go-telegram-bot-api/telegram-bot-api/v5"
	"github.com/miti99/store-scraper-bot-go/internal/config"
	"github.com/miti99/store-scraper-bot-go/internal/repository"
)

// /listgroup — Java ListGroupCommand. Admin-only. Lists authorized groups.
type ListGroupCommand struct {
	cfg       *config.Config
	adminRepo *repository.AdminRepository
}

func NewListGroupCommand(cfg *config.Config, adminRepo *repository.AdminRepository) *ListGroupCommand {
	return &ListGroupCommand{cfg: cfg, adminRepo: adminRepo}
}

func (c *ListGroupCommand) Execute(msg *tgbotapi.Message, sender Sender) {
	if !requireAdminUser(msg.From.ID, msg.Chat.ID, c.cfg, sender) {
		return
	}
	if len(splitArgs(msg.CommandArguments())) != 0 {
		_ = sender.SendMessage(msg.Chat.ID, "Invalid arguments")
		return
	}
	groups, err := c.adminRepo.GetAllGroups()
	if err != nil {
		_ = sender.SendMessage(msg.Chat.ID, "Internal server error")
		return
	}
	if len(groups) == 0 {
		_ = sender.SendMessage(msg.Chat.ID, "No groups found")
		return
	}
	var sb strings.Builder
	sb.WriteString(fmt.Sprintf("<b>Authorized groups (%d):</b>\n", len(groups)))
	for i, gid := range groups {
		sb.WriteString(fmt.Sprintf("%d. <code>%d</code>\n", i+1, gid))
	}
	_ = sender.SendMessage(msg.Chat.ID, sb.String())
}

File: src/bot/commands/list-group.js
import * as adminRepo from '../../repository/admin-repository.js';
import { getCommandArguments, requireAdminUser, splitArgs } from './command-utils.js';

// /listgroup — Java ListGroupCommand. Admin-only.
export function createListGroupCommand(config) {
  return async (msg, sender) => {
    if (!(await requireAdminUser(msg.from.id, msg.chat.id, config, sender))) return;
    if (splitArgs(getCommandArguments(msg.text)).length !== 0) {
      await sender.sendMessage(msg.chat.id, 'Invalid arguments');
      return;
    }
    try {
      const groups = await adminRepo.getAllGroups();
      if (groups.length === 0) {
        await sender.sendMessage(msg.chat.id, 'No groups found');
        return;
      }
      const lines = [`<b>Authorized groups (${groups.length}):</b>`];
      groups.forEach((gid, i) => lines.push(`${i + 1}. <code>${gid}</code>`));
      await sender.sendMessage(msg.chat.id, lines.join('\n') + '\n');
    } catch {
      await sender.sendMessage(msg.chat.id, 'Internal server error');
    }
  };
}
